### PR TITLE
Update version to 27.0.0 and add changelog

### PR DESCRIPTION
**Pull request by alamb**
# Which issue does this PR close?

re https://github.com/apache/arrow-rs/issues/3045

# Rationale for this change
 release

# What changes are included in this PR?

Update version to 27.0.0 and add changelog

See rendered changelog here: https://github.com/alamb/arrow-rs/blob/alamb/prepare_for_27.0.0/CHANGELOG.md

# Are there any user-facing changes?

Update version to 27.0.0 and changelog

### CODE REVIEW

**Comment by iajoiner** [comment]
Ah you are a few minutes faster than me haha.

**Comment by alamb** [comment]
> Ah you are a few minutes faster than me haha.


LOL -- I wake up early. Perhaps we should start these PRs on Thursday 🤔

**Comment by iajoiner** [comment]
@alamb Yup haha

**Review by viirya — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -19,101 +19,121 @@
...
+- Recurse into Dictionary value type in DataType::is\_nested [\#3083](https://github.com/apache/arrow-rs/pull/3083) [[arrow](https://github.com/apache/arrow-rs/labels/arrow)] ([tustvold](https://github.com/tustvold))
+- early type checks in `RowConverter` [\#3080](https://github.com/apache/arrow-rs/pull/3080) [[arrow](https://github.com/apache/arrow-rs/labels/arrow)] ([crepererum](https://github.com/crepererum))
+- Add Decimal128 and Decimal256 to downcast\_primitive [\#3056](https://github.com/apache/arrow-rs/pull/3056) [[arrow](https://github.com/apache/arrow-rs/labels/arrow)] ([viirya](https://github.com/viirya))
+- Validate decimal256 with i256 directly [\#3025](https://github.com/apache/arrow-rs/pull/3025) [[arrow](https://github.com/apache/arrow-rs/labels/arrow)] ([viirya](https://github.com/viirya))
+- Replace hour\_generic with hour\_dyn [\#3006](https://github.com/apache/arrow-rs/pull/3006) [[arrow](https://github.com/apache/arrow-rs/labels/arrow)] ([viirya](https://github.com/viirya))
```
Comment (tustvold) [inline CHANGELOG.md:32]: I think we're missing the other ones related to this, will fix

File: CHANGELOG.md [hunk 2]
```diff
@@ -19,101 +19,121 @@
...
+- Inconsistent Nan Handling Between Scalar and Non-Scalar Comparison Kernels [\#3074](https://github.com/apache/arrow-rs/issues/3074)
+- Debug format for timestamp ignores timezone [\#3069](https://github.com/apache/arrow-rs/issues/3069)
+- Row format decode loses timezone [\#3063](https://github.com/apache/arrow-rs/issues/3063)
+- binary operator produces incorrect result on arrays with resized null buffer [\#3061](https://github.com/apache/arrow-rs/issues/3061)
+- RLEDecoder Panics on Null Padded Pages [\#3035](https://github.com/apache/arrow-rs/issues/3035) [[parquet](https://github.com/apache/arrow-rs/labels/parquet)]
+- Nullif with incorrect valid\_count [\#3031](https://github.com/apache/arrow-rs/issues/3031)
+- RLEDecoder::get\_batch\_with\_dict may panic on bit-packed runs longer than 1024 [\#3029](https://github.com/apache/arrow-rs/issues/3029)
+- Converted type is None according to Parquet Tools then utilizing logical types [\#3017](https://github.com/apache/arrow-rs/issues/3017)
+- CompressionCodec LZ4 incompatible with C++ implementation [\#2988](https://github.com/apache/arrow-rs/issues/2988)
```
Comment (tustvold) [inline CHANGELOG.md:67]: This is backwards compatible but changes what is written, so will mark as api-change

**Review by tustvold — APPROVED** [review]
I've marked a few PRs as api-change, with a regen this should be good to go

**Comment by alamb** [comment]
I am starting to update and plan to cut the RC within the hour.

**Pull request merged**